selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java

```diff
@@ -300,4 +363,83 @@ public void clickOnDeleteButtonInDialogWindow() {
 
     return titles;
   }
+
+  public int getVisibleWorkspacesCount() {
+    return seleniumWebDriverHelper
+        .waitVisibilityOfAllElements(
+            By.xpath("//div[@class='workspace-name-clip']/parent::div/parent::div/parent::div"))
+        .size();
+  }
+
+  public String getFullNameOfWorkspacesListItem(int index) {
+    String itemXpath = String.format(Locators.WORKSPACE_LIST_ITEM, index);
+    String fullNameXpath = itemXpath + "//div[@class='workspace-name-clip']";
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(By.xpath(fullNameXpath), "id")
+        .replace("ws-full-name-", "");
+  }
+
+  public Workspaces.WorkspaceListItem getWorkspacesListItemByWorkspaceName(
+      List<Workspaces.WorkspaceListItem> itemsList, String workspaceName) {
+    return itemsList
+        .stream()
+        .filter(item -> item.getWorkspaceName().equals(workspaceName))
+        .collect(Collectors.toList())
+        .get(0);
+  }
+
+  public List<WorkspaceListItem> getVisibleWorkspaces() {
+    List<WorkspaceListItem> items = new ArrayList<>();
+
+    for (int i = 1; i <= getVisibleWorkspacesCount(); i++) {
+      String fullName = getFullNameOfWorkspacesListItem(i);
+      String ownerName = Arrays.asList(fullName.split("/")).get(0);
+      String workspaceName = Arrays.asList(fullName.split("/")).get(1);
+      int ramCount =
+          Integer.parseInt(Arrays.asList(getWorkspaceRamValue(workspaceName).split(" ")).get(0));
+      int projectsCount = Integer.parseInt(getWorkspaceProjectsValue(workspaceName));
+      items.add(new WorkspaceListItem(ownerName, workspaceName, ramCount, projectsCount));
+    }
+
+    return items;
+  }
+
+  public static class WorkspaceListItem {
+    private String ownerName;
+    private String workspaceName;
+    private int ramAmount;
+    private int projectsAmount;
+
+    public WorkspaceListItem(
+        String ownerName, String workspaceName, int ramAmount, int projectsAmount) {
+      this.ownerName = ownerName;
+      this.workspaceName = workspaceName;
+      this.ramAmount = ramAmount;
+      this.projectsAmount = projectsAmount;
+    }
+
+    public String getOwnerName() {
+      return ownerName;
+    }
+
+    public String getWorkspaceName() {
+      return workspaceName;
+    }
+
+    public int getRamAmount() {
+      return ramAmount;
+    }
+
+    public int getProjectsAmount() {
+      return projectsAmount;
+    }
+
+    public boolean equals(Object obj) {
+      WorkspaceListItem itemForCompare = (WorkspaceListItem) obj;
```

Comment:
May be using Generic more preferable in this case?

Reply:
Generics hardly ever help to simplify the code so as we are overriding `Object#equals(Object)` method here.
But we could make this code safer if we are checking if `obj` an instance of `WorkspaceListItem` as follow https://stackoverflow.com/a/2316239/2556329
And also `@Override` annotation is missed here as well.

@Ohrimenko1988: please, make sure there are no new warnings before merging into the master like the follow:
```
/home/ndp/projects/che/selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/Workspaces.java:437: warning: [EqualsHashCode] Classes that override equals should also override hashCode.
    public boolean equals(Object obj) {
                   ^
    (see http://errorprone.info/bugpattern/EqualsHashCode)
```

Reply:
@dmytro-ndp Thank you for so complete clarification.